From ad94f605004f358386c9c103d17e8dd7b77a94fb Mon Sep 17 00:00:00 2001
From: Ronak Shah <ronak.shah@xilinx.com>
Date: Tue, 1 Sep 2020 07:07:05 -0700
Subject: [PATCH 1549/1852] v4l: xilinx: multi-scaler: Derive the caps from
 video device node

commit b5c056fa30a1771a2b13d2efc6d0f73876272afa from
https://github.com/Xilinx/linux-xlnx.git

This patch derives the V4L2 device capabilities from video device
node instead of hardcoding it.

Signed-off-by: Ronak Shah <ronak.shah@xilinx.com>
Reviewed-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-multi-scaler.c | 11 +++++------
 1 file changed, 5 insertions(+), 6 deletions(-)

diff --git a/drivers/media/platform/xilinx/xilinx-multi-scaler.c b/drivers/media/platform/xilinx/xilinx-multi-scaler.c
index c07dae202358..d40ee9d21e33 100644
--- a/drivers/media/platform/xilinx/xilinx-multi-scaler.c
+++ b/drivers/media/platform/xilinx/xilinx-multi-scaler.c
@@ -1766,18 +1766,17 @@ static int xm2msc_enum_fmt_vid_out(struct file *file, void *fh,
 static int xm2msc_querycap(struct file *file, void *fh,
 			   struct v4l2_capability *cap)
 {
+	struct xm2msc_chan_ctx *chan_ctx = fh_to_chanctx(fh);
+	struct video_device *vfd = &chan_ctx->vfd;
+
 	strncpy((char *)cap->driver, XM2MSC_DRIVER_NAME,
 		sizeof(cap->driver) - 1);
 	strncpy((char *)cap->card, XM2MSC_DRIVER_NAME, sizeof(cap->card) - 1);
 	snprintf((char *)cap->bus_info, sizeof(cap->bus_info),
 		 "platform:%s", XM2MSC_DRIVER_NAME);
-	/*
-	 * This is only a mem-to-mem video device. The STREAMING
-	 * device capability flags are left only for compatibility
-	 * and are scheduled for removal.
-	 */
-	cap->device_caps = V4L2_CAP_VIDEO_M2M_MPLANE;
+	cap->device_caps = vfd->device_caps;
 	cap->capabilities = cap->device_caps | V4L2_CAP_DEVICE_CAPS;
+
 	return 0;
 }
 
-- 
2.31.1

